### ISSUE TITLE
New-NugetPackage fails on german dotnet core (win)

### ISSUE THREAD

**Issue by jmarkert** [PowerShell/PSResourceGet#149]
Steps to reproduce
------------------
Publishing a module to my private PSRepositoy (Nexus3) fails. I'm not sure, but I guess it is broken since I have upgraded PScore from 6.1 to 6.2

```powershell
Publish-Module -Path ".\dest\$ModuleName\$version\" -Repository $MyPrivateRepo -NuGetApiKey $NuGetApiKey -Credential $Credential
```

Actual behavior
---------------

```none
Publish-PSArtifactUtility : Failed to generate the compressed file for module 'Cannot index into a null array.'.
At C:\program files\powershell\6\Modules\PowerShellGet\PSModule.psm1:10785 char:17
+ ...             Publish-PSArtifactUtility @PublishPSArtifactUtility_Param ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidOperation: (:) [Write-Error], WriteErrorException
+ FullyQualifiedErrorId : FailedToCreateCompressedModule,Publish-PSArtifactUtility
```
I debugged the process and found the problem in the New-NugetPackage function
https://github.com/PowerShell/PowerShellGet/blob/7de99ee0c38611556e5c583ffaca98bb1922a0d4/src/PowerShellGet/private/functions/New-NugetPackage.ps1#L119

dotnet.exe returns and localized string, so the -match does not work. For the positive match in german the expression should be something like:
```none
'Das Paket "(.*.nupkg)" wurde erfolgreich erstellt.' 
```
For testing I have changed the expression and the package creation gets recognized correctly. 

Environment data
----------------
I'am on german Windows 10 1903

```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      6.2.1
PSEdition                      Core
GitCommitId                    6.2.1
OS                             Microsoft Windows 10.0.18362
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

```powershell
> Get-Module -ListAvailable PowerShellGet,PackageManagement
    Directory: C:\program files\powershell\6\Modules

ModuleType Version    Name                                PSEdition ExportedCommands
---------- -------    ----                                --------- ----------------
Script     1.3.2      PackageManagement                   Desk      {Find-Package, Get-Package, Get-PackageProvider, Get-PackageSource…}
Script     2.1.3      PowerShellGet                       Desk      {Find-Command, Find-DSCResource, Find-Module, Find-RoleCapability…}
```

```powershell
> Get-PackageProvider -ListAvailable
Name                     Version          DynamicOptions
----                     -------          --------------
NuGet                    3.0.0.1          Destination, ExcludeVersion, Scope, SkipDependencies, Headers, FilterOnTag, Contains, AllowPrereleaseVersions, ConfigFile, SkipValidate
PowerShellGet            2.1.3.0          PackageManagementProvider, Type, Scope, AllowClobber, SkipPublisherCheck, InstallUpdate, NoPathUpdate, AllowPrereleaseVersions, Filter, Tag, Includes, DscResource, RoleCapability, Command, Acce…
```

 
It looks like there is another issue when publishing it to the repo (Publish-NugetPackage). The package gets pushed correctly but an error is returned.

```powershell
...
info : Ihr Paket wurde per Push übertragen.

Failed to publish module 'mymodule': 'nuget.exe failed to push Im nicht interaktiven Modus kann keine Eingabeaufforderung erfolgen.
'.
At C:\program files\powershell\6\Modules\PowerShellGet\PSModule.psm1:10784 char:17
+ ...             Publish-PSArtifactUtility @PublishPSArtifactUtility_Param ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : FailedToPublishTheModule,Publish-PSArtifactUtility
```

**Comment by Stephanevg** [comment]
Can you tell me how I can publish my module today @alerickson ?
I need to publish a module on the gallery, and I am facing this issue. can you point me to:
1) a quickfix?
2) an alternative
please ?

**Comment by alerickson** [comment]
I would suggest using the nuget cli (nuget push) or dotnet cli (dotnet nuget push), the current version of PowerShellGet extends that functionality, so you should be able to use this as a workaround.  Please let me know if this doesn't work for you though.

**Comment by KUTlime** [comment]
@Stephanevg 
1. You can install the latest PowerShellGet v2.x.x, have look at my PR and fix the problem locally at your module instance. It is trivial.
2. I can provide you my custom build of this module which works. I'm using to publishing my PS modules last 5 months because my PR is been ignored for that long. 🤷‍♂️

**Comment by KUTlime** [comment]
@alerickson Are you sure that this issue still remains in PowerShellGet v3? 

@Stephanevg doesn't mention which version he tried to use. It was probably v2.2.3. Since you have rewritten the module into C#, I wouldn't be surprised that the issue is "solved" by this. There is not even `New-NugetPackage` cmdlet in PowerShellGet v3. 🤷‍♀️

**Comment by Stephanevg** [comment]
@KUTlime I really appreciate the offer! But I actually spin up a vagrant windows machine, copied things over to that VM, and pushed it from there. It solved my problem, but clearly not in the most elegant way..

I really hope this module will get better with v3

**Comment by KUTlime** [comment]
@Stephanevg #Me2 but based on quick scouting of the current code, it will be a looong journey. 🤭

**Comment by alerickson** [comment]
@KUTlime I move issues over so that we can ensure we don't replicate problems.  Publish will no longer take a dependency on the dotnet executable, so likely no, but we are using dotnet cli functions. Considering active work is not being done on this repository, it makes more sense to me to move the issue to the repo where it can be properly resolved.

@Stephanevg for future reference, using the dotnet or nuget cli to push should work as well.  Either should already be installed due to PowerShellGet.

@KUTlime We will have a release with 'publish' in the following two weeks, and 'update' following that.  The basic functionality for the  module will be done then.  I'd love to address any bugs, feel free to open up issues in the current repository (or upvote an existing issue) if there's any you've encountered.

**Comment by KUTlime** [comment]
@Stephanevg Honestly? The current C# code has practical issues like

**Comment by KUTlime** [comment]
> @KUTlime free to open up issues in the current repository (or upvote an existing issue) if there's any you've encountered.

I've already did. 😉

Generally, I must conclude that the current code has low quality. If you open the solution, R# give you several hundreds of warnings or code improvements. "Messy" is the word which pop up in my mind when I was checking it.

I understand that it an early stage of development but even then I would pay more attention code quality and consistency. #IMHO

**Comment by alerickson** [comment]
@KUTlime I agree with you--  the goal is to get functionality out and push as quickly as possible so that we can get feedback on user experience.  It certainly is messy, but that's particular because I haven't been focused on implementing one function at a time (though the preview releases are based on new cmdlet functionality).   There's a lot of cleaning that needs to be done, but at this point in time I'm most concerned with someone's experience using the module.  Though I'll definitely have things refactored and cleaned when we GA and start accepting PRs

**Comment by SydneyhSmith** [comment]
Closing this as the publish functionality has been updated/released in PowerShellGet 3.0 previews, thanks everyone for your patience
